From 46ecfa5d5fcad87c7555d5b09569f7b198511adb Mon Sep 17 00:00:00 2001
From: Amit KUMAR <amit-emr.kumar@st.com>
Date: Fri, 28 Mar 2025 15:19:42 -0500
Subject: [PATCH 12/14] dts: add stm32mp257f-dk board support for Android

Update display configuration :
- disable HDMI
- use LVDS by default

Memory :
- add ramoops area

Signed-off-by: Nicolas LOUBOUTIN <nicolas.louboutin@st.com>
---
 .../arm/dts/stm32mp257f-dk-ca35tdcid-resmem.dtsi | 14 +++++++++++++-
 arch/arm/dts/stm32mp257f-dk.dts                  | 16 +++++++++-------
 2 files changed, 22 insertions(+), 8 deletions(-)

diff --git a/arch/arm/dts/stm32mp257f-dk-ca35tdcid-resmem.dtsi b/arch/arm/dts/stm32mp257f-dk-ca35tdcid-resmem.dtsi
index 7e2df35693f..b80552284b2 100644
--- a/arch/arm/dts/stm32mp257f-dk-ca35tdcid-resmem.dtsi
+++ b/arch/arm/dts/stm32mp257f-dk-ca35tdcid-resmem.dtsi
@@ -53,7 +53,15 @@
 		};
 
 		cm33_retram: cm33-retram@a080000 {
-			reg = <0x0 0xa080000 0x0 0x1f000>;
+			reg = <0x0 0xa080000 0x0 0x1b000>;
+			no-map;
+		};
+
+		ramoops_retram: ramoops@a09b000 {
+			compatible = "ramoops";
+			reg = <0x0 0xa09b000 0x0 0x4000>;
+			console-size = <0x1000>;
+			pmsg-size = <0x3000>;
 			no-map;
 		};
 
@@ -81,21 +89,25 @@
 
 		/* DDR reserved memory declaration */
 		tfm_code: tfm-code@80000000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80000000 0x0 0x100000>;
 			no-map;
 		};
 
 		cm33_cube_fw: cm33-cube-fw@80100000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80100000 0x0 0x800000>;
 			no-map;
 		};
 
 		tfm_data: tfm-data@80900000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80900000 0x0 0x100000>;
 			no-map;
 		};
 
 		cm33_cube_data: cm33-cube-data@80a00000 {
+			compatible = "shared-dma-pool";
 			reg = <0x0 0x80a00000 0x0 0x800000>;
 			no-map;
 		};
diff --git a/arch/arm/dts/stm32mp257f-dk.dts b/arch/arm/dts/stm32mp257f-dk.dts
index 0345d346eff..7a3037dbdf9 100644
--- a/arch/arm/dts/stm32mp257f-dk.dts
+++ b/arch/arm/dts/stm32mp257f-dk.dts
@@ -165,14 +165,14 @@
 		compatible = "gpio-backlight";
 		gpios = <&gpioi 7 GPIO_ACTIVE_HIGH>;
 		default-on;
-		default-brightness-level = <0>;
+		default-brightness-level = <1>;
 		status = "okay";
 	};
 
-	sound {
+	sound: sound {
 		compatible = "audio-graph-card";
 		label = "STM32MP25-DK";
-		dais = <&i2s2_port &mdf1_port0 &mdf1_port1>;
+		dais = <&mdf1_port0 &mdf1_port1>;
 		status = "okay";
 	};
 
@@ -241,7 +241,7 @@
 &dsi {
 	vdd-supply =  <&scmi_vddcore>;
 	vdda18-supply = <&scmi_v1v8>;
-	status = "okay";
+	status = "disabled";
 
 	ports {
 		#address-cells = <1>;
@@ -339,7 +339,7 @@
 		compatible = "adi,adv7535";
 		reg = <0x3d>, <0x3c>, <0x3f>, <0x38>;
 		reg-names = "main", "cec", "edid", "packet";
-		status = "okay";
+		status = "disabled";
 		adi,dsi-lanes = <4>;
 		clocks = <&clk_ext_cec>;
 		clock-names = "cec";
@@ -407,7 +407,7 @@
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2s2_pins_b>;
 	pinctrl-1 = <&i2s2_sleep_pins_b>;
-	status = "okay";
+	status = "disabled";
 
 	i2s2_port: port {
 		i2s2_endpoint: endpoint {
@@ -442,9 +442,11 @@
 };
 
 &lvds {
-	status = "okay";
+	default-on;
 	vdd-supply =  <&scmi_vddcore>;
 	vdda18-supply = <&scmi_v1v8>;
+	status = "okay";
+
 	ports {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
2.34.1

